subs/pantry/package.yaml

```diff
@@ -104,6 +107,10 @@ library:
   # For testing
   - Pantry.Internal
   - Pantry.Internal.StaticBytes
+  # For stackage-server
```

Comment:
I'd rather see these in some kind of `.Internal` namespace, I don't want these to be part of the publicly accessed API. Better yet, if there's some way to do it, I'd prefer if there was a minimal subset of these modules that could reasonably be exported from `Pantry` itself.

Reply:
The only thing I can think of is I can compile a list of all functions and types that are needed by stackage-server, and reexport them through a separate module. There are quite a few of them, since I am using pantry's tables directly and esqueleto for queries as well as reuse a bunch of types. Quick compilation of pantry related imports inside stackage-server:

```haskell
import Pantry (defaultHackageSecurityConfig)
import Pantry.Hackage (DidUpdateOccur(..), forceUpdateHackageIndex,
                       getHackageTarballOnGPD)
import Pantry.Storage (HackageCabalId, getTreeForKey, loadBlobById, treeCabal)
import Pantry.Types (CabalFileInfo(..), HpackExecutable(HpackBundled),
                     PackageIdentifierRevision(..), PantryConfig(..),
                     packageTreeKey)
```
```haskell
import Pantry.SHA256 (fromHexText)
import Pantry.Storage (BlobId, HackageCabalId)
import Pantry.Types (BlobKey(..), CabalFileInfo(..), FileSize(..),
                     HasPantryConfig(..), PackageIdentifierRevision(..),
                     PackageNameP(..), PantryConfig, TreeKey(..), VersionP(..))
```

```haskell
import Pantry.Storage as PS (BlobId, HackageCabalId, ModuleNameId,
                             PackageNameId, Tree(..), TreeEntry(..),
                             TreeEntryId, TreeId, Unique(..), VersionId,
                             unBlobKey)
import qualified Pantry.Storage as Pantry (migrateAll)
import Pantry.Types (HasPantryConfig(..), PantryConfig(..), Revision,
                     Storage(..))
```
```haskell
import Pantry.Storage (EntityField(..), PackageName, Unique(..), Version,
                       getBlobKey, getPackageNameById, getPackageNameId,
                       getTreeForKey, getVersionId, loadBlobById, treeCabal)
import Pantry.Types (mkSafeFilePath)
```

Reply:
Alright, let's do reexporting from a `.Internal` module with a comment explaining that this is for advanced usages.

Reply:
I created an extra module `Pantry.Internal.Stackage` which exports only the stuff needed for stackage server.